File: C_ project/Bubble/Timepass /BubbleSort.py
def display(arr):
  print(" ".join(map(str, arr)))

def main():
  n = int(input("Enter the value of n:\n"))
  a = []
  print("Enter", n, "elements:")
  for i in range(n):
      a.append(int(input()))

  print()

  # Computation of bubble sort
  for pass_num in range(1, n):
      print(f"pass {pass_num}:")
      for j in range(n - pass_num):
          if a[j] > a[j + 1]:
              a[j], a[j + 1] = a[j + 1], a[j]  # Swap
          display(a)
      print()

  print("The sorted array after Bubble sort is:")
  display(a)

if __name__ == "__main__":
  main()

File: C_ project/Bubble/Timepass /BubbleSort.java
import java.util.Scanner;

public class BubbleSort {

    public static void display(int[] a, int n) {
        for (int i = 0; i < n; i++) {
            System.out.print(a[i] + " ");
        }
        System.out.println();
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.println("Enter the value of n:");
        int n = scanner.nextInt();
        int[] a = new int[n];

        System.out.println("Enter " + n + " elements:");
        for (int i = 0; i < n; i++) {
            a[i] = scanner.nextInt();
        }
        System.out.println();

        // Computation of bubble sort
        for (int pass = 1; pass <= n - 1; pass++) {
            System.out.println("pass " + pass + ":");
            for (int j = 0; j < n - pass; j++) {
                if (a[j] > a[j + 1]) {
                    int temp = a[j + 1];
                    a[j + 1] = a[j];
                    a[j] = temp;
                }
                display(a, n);
            }
            System.out.println();
        }

        System.out.println("The sorted array after Bubble sort is:");
        display(a, n);
    }
}
